example/command_line_example/osm_test.md

```diff
@@ -0,0 +1,42 @@
+## Example to test implementation for reading OSM data directly
+
+These instructions were tested on Ubuntu 18.04, after building the code in the OSM branch. They assume a working directory under ``example/command_line_example/``
+
+### 1. Download test data
+
+```
+mkdir osm_test
+cd osm_test
+wget https://www.dropbox.com/s/a3d8hmtkflckw10/fmm_osm_test_data.zip
```

Comment:
Yes, I meant this as a quick test, for a case where I know that the shapefile and the OSM file should give the exact same result.

I agree it is not suitable for example code.

I suggest that the example code can only work on OSM data and leave out the shapefile part.

We could additionally have a more complete test case with instructions to download OSM data and either extract a shapefile with osmnx or use the OSM data directly. It could be a good to have a testcase that is expected to work in general and give exact same output, so that the OSM reader code can be validated.

Since doing the test case in the Dropbox, I changed the filters to match the behavior of osmnx, so now a test based on an osmnx shapefile should be feasible.

Reply:
Actually I have created a separate repo https://github.com/cyang-kth/osm_mapmatching for instructions about working with OSM data where I use OSMNX and PostGIS for preprocessing. 

Perhaps some command-line instructions to process osm file should also be placed there. A link can be provided here.